client/src/modules/account_reference_type/account_reference_type.js

```diff
@@ -0,0 +1,103 @@
+angular.module('bhima.controllers')
+  .controller('AccountReferenceTypeController', AccountReferenceTypeController);
+
+AccountReferenceTypeController.$inject = [
+  '$state', 'AccountReferenceTypeService', 'NotifyService', 'uiGridConstants', 'ModalService', '$translate',
+];
+
+/**
+ * Account Reference Type Controller
+ * This module is responsible for handling the CRUD operation on Account Reference Type
+ */
+
+function AccountReferenceTypeController($state, AccountReferenceType, Notify, uiGridConstants,
+  ModalService, $translate) {
+  const vm = this;
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+  vm.toggleFilter = toggleFilter;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      {
+        field : 'label',
+        displayName : 'FORM.LABELS.DESIGNATION',
+        enableFiltering : 'true',
+        headerCellFilter : 'translate',
+      }, {
+        field : 'action',
+        displayName : '',
+        width : 120,
+        enableFiltering : 'false',
+        cellTemplate : '/modules/account_reference_type/templates/action.cell.html',
+      },
+    ],
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  }
+
+  // bind methods
+  vm.edit = edit;
+  vm.remove = remove;
+
+  function edit(accountReferenceType) {
+    $state.go('account_reference_type.edit', { id : accountReferenceType.id });
+  }
+
+  function remove(id) {
+    ModalService.confirm('FORM.DIALOGS.CONFIRM_DELETE')
+      .then((bool) => {
+        if (!bool) { return; }
+
+        AccountReferenceType.delete(id)
+          .then(() => {
+            Notify.success('FORM.INFO.DELETE_SUCCESS');
+            loadGrid();
+          })
+          .catch(Notify.handleError);
+      });
+  }
+
+  function handleError(error) {
+    vm.hasError = true;
+    Notify.handleError(error);
+  }
+
+  function loadGrid() {
+    toggleLoadingIndicator();
+    vm.hasError = false;
+
+    AccountReferenceType.read()
+      .then((accountReferenceType) => {
+        accountReferenceType.forEach((item) => {
+          item.label = $translate.instant(item.label);
```

Comment:
It looks like every time we download account reference types, we want to translate them.  Let's move this operation into the `AccountReferenceTypeService` so that after every `read()` call, we translate these labels.